vyper/codegen/stmt.py

```diff
@@ -147,15 +147,16 @@ def _assert_reason(self, test_expr, msg):
             revert_seq = [
                 "seq",
                 ["mstore", buf, method_id],
-                ["revert", add_ofst(buf, 28), ["add", 4, encoded_length]],
             ]
+            revert = ["revert", add_ofst(buf, 28), ["add", 4, encoded_length]]
+            revert_seq.append(IRnode.from_list(revert, error_msg="user revert with reason"))
```

Comment:
i think maybe it will be cleaner to have something which propagates error_msg down to children

Reply:
i think we have `IRNode.set_error_msg` for this purpose. maybe it should be part of the constructor or something :thinking:

Reply:
When we assign the error message to multiple nodes, it appears multiple times in the source map, each associated with a different PC value. I assume, we only want the error message to appear once - which PC value is the correct one?

Reply:
it's ok to appear multiple times in the PC map to be honest, it's just metadata for tooling. and since we can't guarantee that there is a one-to-one correspondence between source locations and PCs (because of optimizations), it's ok for the error message to appear multiple times.